### PR TITLE
A few more tweaks

### PR DESCRIPTION
**Pull request by benbrittain**
removed `</p>`, also added some ZSTs to prevent the need to panic

### CODE REVIEW

**Comment by ninjasource** [comment]
Thanks for this! I'll be able to take a look later today.

**Comment by benbrittain** [comment]
It mildly complicates the type signatures, totally understandable if not merged. I'm just a big fan of avoiding panics where possible.

I was also thinking about how much of this could be made less manual with the right trait. Haven't thought about it too much though because I got distracted by a different project :P

File: examples/client.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ impl From<Utf8Error> for WebClientError {
 pub fn template_client() -> Result<()> {
     let mut buffer1: [u8; 1000] = [0; 1000];
     let mut buffer2: [u8; 1000] = [0; 1000];
-    let mut websocket = ws::WebSocket::new_client(rand::thread_rng());
+    let mut websocket = ws::WebSocket::<_, ws::Client>::new_client(rand::thread_rng());
```
Comment (ninjasource) [inline examples/client.rs:51]: ```WebSocket::<_, ws::Client>``` can be replaced with ```WebSocketClient```

File: src/lib.rs [hunk 2]
```diff
@@ -388,19 +410,12 @@ where
...
-    /// # Panics
-    /// Panics with `Server websocket expected` if the user calls this function as a client
-    /// websocket. This function should be use with a websocket created using `new_server()`
     pub fn server_accept(
         &mut self,
         sec_websocket_key: &WebSocketKey,
         sec_websocket_protocol: Option<&WebSocketSubProtocol>,
         to: &mut [u8],
     ) -> Result<usize> {
-        if self.is_client {
```
Comment (ninjasource) [inline src/lib.rs:401]: Really great that you got rid of the need for this. There is still a panic in the client_accept function that can now be removed too.

**Comment by ninjasource** [comment]
Thank you so much for this PR. I think that getting rid of the need to panic using PhantomData is really neat. I'm going to merge it and and make the changes I mentioned above.

**Pull request merged**